File: pulse-java/src/main/java/com/fluxchess/pulse/model/CastlingType.java
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse.model;

public final class CastlingType {

	public static final int KINGSIDE = 0;
	public static final int QUEENSIDE = 1;

	public static final int NOCASTLINGTYPE = 2;

	public static final int[] values = {
			KINGSIDE, QUEENSIDE
	};

	private CastlingType() {
	}
}

File: pulse-go/internal/pulse/castling_type.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

type CastlingType = int

const (
	Kingside  CastlingType = 0
	Queenside CastlingType = 1

	NoCastlingType CastlingType = 2
)

var (
	CastlingTypes = [2]CastlingType{Kingside, Queenside}
)
